### PR TITLE
CTCP-4873 Add rows for commodity under House Consignment / Item

### PR DESCRIPTION
**Pull request by karahanozturk**
https://jira.tools.tax.service.gov.uk/browse/CTCP-4873

image

### CODE REVIEW

File: app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala [hunk 1]
```diff
@@ -57,4 +57,31 @@ class ConsignmentItemAnswersHelper(
     id = Some(s"change-net-weight-${houseConsignmentIndex.display}"),
     call = Some(Call(GET, "#"))
   )
+
+  def cusCodeRow(houseConsignmentIndex: Index, itemIndex: Index): Option[SummaryListRow] = getAnswerAndBuildRow[String](
```
Comment (jackheslop96) [inline app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala:61]: Don't need to pass the indexes here. They're available in the class definition

File: app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala [hunk 2]
```diff
@@ -57,4 +57,31 @@ class ConsignmentItemAnswersHelper(
...
+    page = CustomsUnionAndStatisticsCodePage(houseConsignmentIndex, itemIndex),
+    formatAnswer = formatAsText,
+    prefix = "unloadingFindings.rowHeadings.item.cusCode",
+    args = itemIndex.display,
+    id = Some(s"change-cus-code-${houseConsignmentIndex.display}"),
+    call = Some(Call(GET, "#"))
+  )
+
+  def commodityCodeRow(houseConsignmentIndex: Index, itemIndex: Index): Option[SummaryListRow] = getAnswerAndBuildRow[String](
```
Comment (jackheslop96) [inline app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala:70]: As above

File: app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala [hunk 3]
```diff
@@ -57,4 +57,31 @@ class ConsignmentItemAnswersHelper(
...
+    page = CommodityCodePage(houseConsignmentIndex, itemIndex),
+    formatAnswer = formatAsText,
+    prefix = "unloadingFindings.rowHeadings.item.commodityCode",
+    args = itemIndex.display,
+    id = Some(s"change-commodity-code-${houseConsignmentIndex.display}"),
+    call = Some(Call(GET, "#"))
+  )
+
+  def nomenclatureCodeRow(houseConsignmentIndex: Index, itemIndex: Index): Option[SummaryListRow] = getAnswerAndBuildRow[String](
```
Comment (jackheslop96) [inline app/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelper.scala:79]: As above

File: test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala [hunk 4]
```diff
@@ -111,4 +114,161 @@ class ConsignmentItemAnswersHelperSpec extends AnswersHelperSpecBase {
...
+                    |              {
+                    |                  "Commodity" : {
+                    |                      "cusCode" : "$cusCode"
+                    |                  }
+                    |              }
+                    |          ]
+                    |        }
+                    |      ]
+                    |    }
+                    |}
```
Comment (jackheslop96) [inline test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala:150]: Can we set the values in user answers instead? Defining the answers like this makes these tests very dependent on the paths in the page objects

File: test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala [hunk 5]
```diff
@@ -111,4 +114,161 @@ class ConsignmentItemAnswersHelperSpec extends AnswersHelperSpecBase {
...
+                    |                          "harmonizedSystemSubHeadingCode" : "$commodityCode"
+                    |                      }
+                    |                  }
+                    |              }
+                    |          ]
+                    |        }
+                    |      ]
+                    |    }
+                    |}
+                    |""".stripMargin)
```
Comment (jackheslop96) [inline test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala:204]: As above

File: test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala [hunk 6]
```diff
@@ -111,4 +114,161 @@ class ConsignmentItemAnswersHelperSpec extends AnswersHelperSpecBase {
...
+                    |                          "combinedNomenclatureCode" : "$nomenclatureCode"
+                    |                      }
+                    |                  }
+                    |              }
+                    |          ]
+                    |        }
+                    |      ]
+                    |    }
+                    |}
+                    |""".stripMargin)
```
Comment (jackheslop96) [inline test/utils/answersHelpers/consignment/houseConsignment/ConsignmentItemAnswersHelperSpec.scala:257]: As above

**Review by rajukakumanu86 — APPROVED** [review]

**Pull request merged**